mrmap/quality/views.py

```diff
@@ -5,88 +5,49 @@
-                            status=status.HTTP_400_BAD_REQUEST)
-
-    user = request.user
-    owned_by_org = metadata.owned_by_org
-
-    success_callback = complete_validation.s()
-    error_callback = complete_validation_error.s(user_id=user.id,
-                                                 config_id=config_id,
-                                                 metadata_id=metadata.id)
-
-    pending_task = run_quality_check.apply_async(args=(config_id, metadata_id),
-                                                 link=success_callback,
-                                                 link_error=error_callback)
-
-    if current_task:
-        current_task.update_state(
-            state=states.STARTED,
-            meta={
-                "current": 10,
-                "phase": f"Validating {metadata.title}",
-            }
-        )
-
-    if current_view is not None:
-        if current_view_arg is not None:
-            return HttpResponseRedirect(
-                reverse(current_view, args=(current_view_arg,)), status=303)
-        else:
-            return HttpResponseRedirect(reverse(current_view), status=303)
-
-    return HttpResponse(status=status.HTTP_200_OK)
-
-
-def get_latest(request, metadata_id: str):
-    metadata = get_object_or_404(Metadata, pk=metadata_id)
-
-    try:
-        latest_run = ConformityCheckRun.objects.get_latest_check(metadata)
-        latest = {
-            "passed": latest_run.passed,
-            "running": latest_run.is_running(),
-        }
-    except ConformityCheckRun.DoesNotExist:
-        latest = {
-            "passed": None,
-            "running": None,
-        }
-
-    return JsonResponse(latest)
+from quality.tables import ConformityCheckRunTable
+from quality.tasks import run_conformity_check
+
+
+class ConformityCheckRunListView(SecuredListMixin, FilterView):
+    model = ConformityCheckRun
+    table_class = ConformityCheckRunTable
+
+
+class ConformityCheckRunCreateView(SecuredCreateView):
+    model = ConformityCheckRun
+    form_class = ConformityCheckRunModelForm
+
+    def form_valid(self, form):
+        response = super().form_valid(form)
+        transaction.commit()
+        org_pk = form.cleaned_data["metadata"].owned_by_org_id
+        job_pk = run_conformity_check(self.object.pk, **{"created_by_user_pk": self.request.user.pk,
+                                                         "owned_by_org_pk": org_pk})
+        try:
+            job = Job.objects.get(pk=job_pk)
+            return HttpResponseRedirect(job.get_absolute_url())
+        except Job.ObjectDoesNotExist:
+            pass
+        return response
+
+
+class ConformityCheckRunDeleteView(SecuredDeleteView):
+    model = ConformityCheckRun
+
+# TODO should this be secured?
+# TODO handle mime type (can currently be HTML or JSON)
+# TODO should this be based on a Class-based View?
+def conformity_check_report_download(request, pk):
```

Comment:
why no class based view? use DetailView instead of function based view.

```python
class ConformityCheckRunResult(DetailView):
    model = ConformityCheckRun
    queryset = ConfirmityCheckRun.objects.filter(result__is_null=False).only("result)
    content_type = "application/xml"  # example

    def render_to_response(context, **kwargs):
       return HttpResponse(self.object.result)
```

Reply:
Done. Additionally, we added a new field for determining the report type (HTML/JSON), so the browser gets the correct content type header.